dotCMS/src/main/java/com/dotmarketing/quartz/job/CascadePermissionsJob.java

```diff
@@ -86,34 +80,21 @@ public CascadePermissionsJob() {
 	}
     
 	public static void triggerJobImmediately (Permissionable perm, Role role) {
-		String randomID = UUID.randomUUID().toString();
-		String userId = null;
-		JobDataMap dataMap = new JobDataMap();
-		
-		dataMap.put("permissionableId", perm.getPermissionId());
-		dataMap.put("roleId", role.getId());
-
-		//TODO: For a major release, remove this logic and get userId as parameter
-		if (UtilMethods.isSet(HttpServletRequestThreadLocal.INSTANCE.getRequest())) {
-			userId = WebAPILocator.getUserWebAPI()
-					.getLoggedInUser(HttpServletRequestThreadLocal.INSTANCE.getRequest())
-					.getUserId();
-		}
-		dataMap.put("userId", userId);
-
-		JobDetail jd = new JobDetail("CascadePermissionsJob-" + randomID, "cascade_permissions_jobs", CascadePermissionsJob.class);
-		jd.setJobDataMap(dataMap);
-		jd.setDurability(false);
-		jd.setVolatility(false);
-		jd.setRequestsRecovery(true);
-		
-		long startTime = System.currentTimeMillis();
-		SimpleTrigger trigger = new SimpleTrigger("permissionsCascadeTrigger-"+randomID, "cascade_permissions_triggers",  new Date(startTime));
-		
 		try {
-			Scheduler sched = QuartzUtils.getSequentialScheduler();
-			sched.scheduleJob(jd, trigger);
-		} catch (SchedulerException e) {
+			String userId = null;
+			if (UtilMethods.isSet(HttpServletRequestThreadLocal.INSTANCE.getRequest())) {
+				userId = WebAPILocator.getUserWebAPI()
+						.getLoggedInUser(HttpServletRequestThreadLocal.INSTANCE.getRequest())
+						.getUserId();
+			}
+			final ImmutableMap<String, Serializable> nextExecutionData = ImmutableMap
+					.of("permissionableId", perm.getPermissionId(),
+							"roleId", role.getId(),
+							"userId", userId
+					);
+
+			DotStatefulJob.enqueueTrigger(nextExecutionData, CascadePermissionsJob.class);
+		} catch (Exception e) {

```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6152219)